homeassistant/components/withings/config_flow.py

```diff
@@ -1,192 +1,64 @@
     """Handle a config flow."""
 
-    VERSION = 1
+    DOMAIN = const.DOMAIN
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+    _current_data = None
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict:
+        """Extra data that needs to be appended to the authorize url."""
+        return {
+            "scope": ",".join(
+                [
+                    AuthScope.USER_INFO.value,
+                    AuthScope.USER_METRICS.value,
+                    AuthScope.USER_ACTIVITY.value,
+                ]
+            )
+        }
 
-    def __init__(self):
-        """Initialize flow."""
-        self.flow_profile = None
-        self.data = None
-
-    def async_profile_config_entry(self, profile: str) -> Optional[ConfigEntry]:
-        """Get a profile config entry."""
-        entries = self.hass.config_entries.async_entries(const.DOMAIN)
-        for entry in entries:
-            if entry.data.get(const.PROFILE) == profile:
-                return entry
-
-        return None
-
-    def get_auth_client(self, profile: str):
-        """Get a new auth client."""
-        flow = self.hass.data[DATA_FLOW_IMPL]
-        client_id = flow[const.CLIENT_ID]
-        client_secret = flow[const.CLIENT_SECRET]
-        base_url = flow[const.BASE_URL].rstrip("/")
-
-        callback_uri = "{}/{}?flow_id={}&profile={}".format(
-            base_url.rstrip("/"),
-            const.AUTH_CALLBACK_PATH.lstrip("/"),
-            self.flow_id,
-            profile,
-        )
-
-        return withings.WithingsAuth(
-            client_id,
-            client_secret,
-            callback_uri,
-            scope=",".join(["user.info", "user.metrics", "user.activity"]),
-        )
-
-    async def async_step_import(self, user_input=None):
-        """Create user step."""
-        return await self.async_step_user(user_input)
-
-    async def async_step_user(self, user_input=None):
-        """Create an entry for selecting a profile."""
-        flow = self.hass.data.get(DATA_FLOW_IMPL)
+    async def async_oauth_create_entry(self, data: dict) -> dict:
+        """Override the create entry so user can select a profile."""
+        self._current_data = data
+        return await self.async_step_profile(data)
 
-        if not flow:
-            return self.async_abort(reason="no_flows")
+    async def async_step_profile(self, data: dict) -> dict:
+        """Prompt the user to select a user profile."""
+        profile = data.get(const.PROFILE)
 
-        if user_input:
-            return await self.async_step_auth(user_input)
+        if profile:
+            new_data = {**self._current_data, **{const.PROFILE: profile}}
+            self._current_data = None
+            return await self.async_step_finish(new_data)
 
+        profiles = self.hass.data[const.DOMAIN][const.CONFIG][const.PROFILES]
```

Comment:
This is not defined if no entry in configuration.yaml.

Reply:
Correct. However in `async_setup`, if the config isn't set, then the config_flow will not be registered.

Reply:
PR incoming 😉